### ISSUE TITLE
Data exceeds the local storage quota of 5MB

### ISSUE THREAD

**Issue by ThisNils** [davidbmaier/discord-recap#6]
After fully loading the statistics, this error is being thrown:
<img src="https://media.discordapp.net/attachments/703190611125076010/932674868288319558/unknown.png">

**Comment by davidbmaier** [comment]
Hmm, interesting - which browser is that? Might be worth trying with a different one just to see how big it really becomes (or trying to adjust the active browser's quota if possible).

Since I don't know anything about your data, do you think this is realistic? Would you say you're a very active user over several years?
For reference, I personally have ~70k messages over 100+ DMs and 100+ servers, and my stats are around 2MB. From what I've seen from others, nobody else seems to have run into this but it's certainly possible to get into that range.

Might have to split the stats into smaller packages, it's tough to test this if you only have one dataset 😅.

**Comment by ThisNils** [comment]
I'm using chrome and the local storage max size is 5MB (as specified in the HTML5 spec). There is no way to change that in chrome (or any other chromium browser) unfortunately.

My zipped data package is 115MB. Not sure if thats a lot though. I've been using discord for a few years now

**Comment by davidbmaier** [comment]
Yeah, the data package is much larger, but that's pretty common cause it contains a ton of garbage.

Just read about Chrome's limitation there - the more you know...
I have some ideas how I could get around that, will update the issue when I've looked into the details.

**Comment by davidbmaier** [comment]
Alright, I had a closer look and decided to switch from `localStorage` to `IndexedDB` which should be way more scalable. It depends a bit more on browser implementation but generally the max size is more bound to actual physical size limitations as opposed to a randomly defined number.

I've tried it locally with sizes up to 25MB but generally this should support way more.
Updated it in https://github.com/davidbmaier/discord-recap/commit/0a59161a46c105939d57f9af2cc5b63278ec63af, feel free to test it on the live version and let me know if that fixes it for you.
